modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/InfrastructureTypeMapping.java

```diff
@@ -36,6 +36,7 @@
       .put(TlsContextFactory.class, new InfrastructureType(TLS_PARAMETER_NAME, 8))
       .put(SourceTransactionalAction.class, new InfrastructureType(TRANSACTIONAL_ACTION_PARAMETER_NAME, 6))
       .put(OperationTransactionalAction.class, new InfrastructureType(TRANSACTIONAL_ACTION_PARAMETER_NAME, 7))
+      .put(TransactionType.class, new InfrastructureType(TRANSACTIONAL_TYPE_PARAMETER_NAME, 9))
```

Comment:
you also need to add tests in the Dsl and application model apis to make sure that we're able to serialize/deserialize an application using this feature to/from the DSL. Look at https://github.com/mulesoft/mule/commit/9c2efa45d7f4d30a942c17cf83bff7b7a32feafa for examples